api/v1alpha2/conversion.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	v1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/deprecated/v1alpha1"
+)
+
+//nolint
+func Convert_v1alpha2_ClusterSpec_To_v1alpha1_ClusterSpec(in *ClusterSpec, out *v1alpha1.ClusterSpec, s conversion.Scope) error {
+	if err := Convert_v1alpha2_ClusterNetworkingConfig_To_v1alpha1_ClusterNetworkingConfig(in.ClusterNetwork, &out.ClusterNetwork, s); err != nil {
+		return err
+	}
+
+	// DISCARDS:
+	// InfrastructureRef
+
+	return nil
+}
+
+//nolint
+func Convert_v1alpha1_ClusterSpec_To_v1alpha2_ClusterSpec(in *v1alpha1.ClusterSpec, out *ClusterSpec, s conversion.Scope) error {
+
+	var outNetworkConfig ClusterNetworkingConfig
+
+	if err := Convert_v1alpha1_ClusterNetworkingConfig_To_v1alpha2_ClusterNetworkingConfig(&in.ClusterNetwork, &outNetworkConfig, s); err != nil {
```

Comment:
Could you put this straight into `&out.ClusterNetwork`?

Reply:
It's a pointer, so out.ClusterNetwork is nil by default

Reply:
Got it, misread. Thanks.